src/HTMLMesh.js

```diff
@@ -494,11 +494,13 @@ function html2canvas( element ) {
 
 				}
 
-				if ( element.type === 'color' || element.type === 'text' || element.type === 'number' ) {
+				if ( element.type === 'color' || element.type === 'text' || element.type === 'number' || element.type === 'email' || element.type === 'password' ) {
 
 					clipper.add( { x: x, y: y, width: width, height: height } );
 
-					drawText( style, x + parseInt( style.paddingLeft ), y + parseInt( style.paddingTop ), element.value );
+					const displayValue = element.type === 'password' ? '*'.repeat(element.value.length) : element.value;
```

Comment:
it should have space inside repeat

Reply:
I added the spacing, kindly check my recent commit.